front/lib/api/auth0.ts

```diff
@@ -1,13 +1,38 @@
+import type { Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
 import { ManagementClient } from "auth0";
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
 import jwt from "jsonwebtoken";
 import jwksClient from "jwks-rsa";
+import type { NextApiRequest } from "next";
 
 import config from "@app/lib/api/config";
 import { UserResource } from "@app/lib/resources/user_resource";
 import logger from "@app/logger/logger";
 
 let auth0ManagemementClient: ManagementClient | null = null;
 
+const Auth0JwtPayloadSchema = t.type({
+  azp: t.string,
+  exp: t.number,
+  scope: t.string,
+  sub: t.string,
+});
+
+export type Auth0JwtPayload = t.TypeOf<typeof Auth0JwtPayloadSchema> &
+  jwt.JwtPayload;
+
+export function getRequiredScope(
+  req: NextApiRequest,
+  requiredScopes?: Record<string, string>
+) {
+  if (requiredScopes && req.method && requiredScopes[req.method]) {
+    return [requiredScopes[req.method]];
```

Comment:
Why do we need to return an array here? If it's expected maybe it should be `getRequiredScopes` and let's define the return type.

Reply:
Note: are they all required ? (usually it's one of them, not all of them)

Reply:
If one of them, i would rename as getAllowedScopes()

Reply:
right, I don't think we need an array for now

Reply:
i was expecting all required, but yes could be an array of "at least one", that's probably more useful.  For now i think we can stick with one single value